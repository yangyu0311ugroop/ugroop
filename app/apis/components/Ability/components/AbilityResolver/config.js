import { ABILITY_DATA_STORE } from 'appConstants';

export const CONFIG = {
  value: {
    tour: ({ nodeId }) => [ABILITY_DATA_STORE, 'tours', nodeId],
    organisation: ({ orgId }) => [ABILITY_DATA_STORE, 'organisation', orgId],
    executeAbilityUpdate: [ABILITY_DATA_STORE, 'executeAbilityUpdate'],
  },
  setValue: {
    abilityUpdated: [ABILITY_DATA_STORE, 'abilityUpdated'],
    executeAbilityUpdate: [ABILITY_DATA_STORE, 'executeAbilityUpdate'],
  },
};
