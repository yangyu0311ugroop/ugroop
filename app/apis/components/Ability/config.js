import { ABILITY_NORMALISERS } from 'apis/components/Ability/normalisers';
import { ABILITY_API, FIND_MY_ABILITIES, FIND_MY_TOURS } from 'apis/constants';
import {
  ABILITY_DATA_STORE,
  ORGANISATION_DATA_STORE,
  USER_DATA_STORE,
} from 'appConstants';
import { FILE_STORE_SELECTORS } from 'datastore/fileStore/selectors';
import { NODE_STORE_SELECTORS } from 'datastore/nodeStore/selectors';
import { ORGANISATION_SCHEMA } from 'datastore/orgStore/schema';
import { USER_STORE_SELECTORS } from 'datastore/userStore/selectors';
import { DATASTORE_UTILS } from 'datastore';
import { first } from 'lodash';
import { normalize } from 'normalizr';
import { requests } from 'utils/request';
import { TOUR_ROLE } from './roles';

const reduce = (definitions, type, key, notSetRole) => (object, element) => ({
  ...object,
  [element[key]]: definitions[type][element.role || notSetRole],
});

export const utility = { reduce };

export const CONFIG = {
  name: ABILITY_API,

  requests: {
    [FIND_MY_ABILITIES]: () =>
      requests.fetchWithAuthorisation('get', `/${ABILITY_API}/me`),
    [FIND_MY_TOURS]: () =>
      requests.fetchWithAuthorisation('get', `/${ABILITY_API}/tours`),
  },

  value: {
    id: USER_STORE_SELECTORS.userId,
  },

  setValue: {
    fetched: [ABILITY_DATA_STORE, 'fetched'],
    definitions: [ABILITY_DATA_STORE, 'definitions'],
    tours: [ABILITY_DATA_STORE, 'tours'],
    organisation: [ABILITY_DATA_STORE, 'organisation'],
    organisations: [ORGANISATION_DATA_STORE, 'organisations'],
    orgUsers: [ORGANISATION_DATA_STORE, 'orgUsers'],
    organisationRoles: [ORGANISATION_DATA_STORE, 'organisations'],
    organisationNode: [ORGANISATION_DATA_STORE, 'organisationNode'],
    organisationTours: [ORGANISATION_DATA_STORE, 'organisationTours'],
    organisationId: [ORGANISATION_DATA_STORE, 'organisationId'],
    organisationIds: [ORGANISATION_DATA_STORE, 'organisationIds'],
    userId: [USER_DATA_STORE, 'userId'],
    files: FILE_STORE_SELECTORS.files,
    users: USER_STORE_SELECTORS.users,
    node: NODE_STORE_SELECTORS.nodes,
    userNode: [USER_DATA_STORE, 'userNode'],
    sortedOrganisationIds: USER_STORE_SELECTORS.organisations,
    preferences: [ORGANISATION_DATA_STORE, 'preferences'],
  },

  processResult: {
    [FIND_MY_ABILITIES]: ({
      id,
      definitions,
      organisation,
      ownedTours,
      sharedTours,
      organisationNode,
      organisationData,
    }) => {
      const ownAsOwnedTours = ownedTours.filter(tour => !tour.role); // Admins & Owner
      const ownedHasRoleTours = ownedTours.filter(tour => !!tour.role); // Other Role
      ownedTours.filter(tour => !tour.role);

      const ownToursAbility = ownAsOwnedTours.reduce(
        utility.reduce(definitions, 'tour', 'id', TOUR_ROLE.TOUR_OWNER),
        {},
      );

      const ownToursOrgAbility = ownedHasRoleTours.reduce(
        utility.reduce(definitions, 'organisation', 'id', ''),
        {},
      );

      const sharedToursAbility = sharedTours.reduce(
        utility.reduce(definitions, 'tour', 'nodeId', TOUR_ROLE.TOUR_OWNER),
        {},
      );

      const { entities, result: organisationIds } = normalize(
        organisationNode,
        ORGANISATION_SCHEMA.organisationNodes,
      );
      const { entities: dataEntities } = normalize(
        organisationData,
        ORGANISATION_SCHEMA.organisationDatas,
      );
      const { entities: rolesEntities } = normalize(
        organisation,
        ORGANISATION_SCHEMA.organisationRoles,
      );
      // hack: currently one user can only belong to one organisation => always the first one
      const organisationId = first(organisationIds);

      return {
        userId: id,
        definitions,
        tours: {
          ...ownToursOrgAbility,
          ...sharedToursAbility,
          ...ownToursAbility,
        }, // Prioritise ownership, shared, org members
        organisation: organisation.reduce(
          utility.reduce(definitions, 'organisation', 'orgId'),
          {},
        ),
        organisationNode: entities.organisationNode,
        organisationId,
        organisationIds,
        organisations: DATASTORE_UTILS.upsertObject(
          dataEntities.organisationData,
        ),
        orgUsers: DATASTORE_UTILS.upsertObject(rolesEntities.organisationRole),
        organisationRoles: DATASTORE_UTILS.upsertObject(
          rolesEntities.organisationRole,
        ),
        files: DATASTORE_UTILS.upsertObject({
          ...dataEntities.photo,
        }),
        users: DATASTORE_UTILS.upsertObject(id, {
          organisations: organisationIds,
        }),
        preferences: DATASTORE_UTILS.upsertObject({
          ...dataEntities.preference,
        }),
        fetched: Date.now(),
      };
    },
    [FIND_MY_TOURS]: ABILITY_NORMALISERS.findMyTours,
  },

  manuallySubscribe: true, // will not auto-subscribe to all values in `@@values`
};
