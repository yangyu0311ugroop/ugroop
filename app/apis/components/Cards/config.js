import {
  CARDS_API,
  CREATE_CUSTOMER_CARD,
  UPDATE_CUSTOMER_CARD,
  DELETE_CUSTOMER_CARD,
  GET_CUSTOMER_CARDS,
} from 'apis/constants';
import { CARD_NORMALISER } from 'apis/components/Cards/helpers';
import { requests } from 'utils/request';

export const CONFIG = {
  name: CARDS_API,
  requests: {
    [GET_CUSTOMER_CARDS]: ({ id }) =>
      requests.fetchWithAuthorisation('get', `/${CARDS_API}/${id}/sources`),
    [CREATE_CUSTOMER_CARD]: ({ id, data }) =>
      requests.fetchWithAuthorisation('post', `/${CARDS_API}/${id}`, data),
    [UPDATE_CUSTOMER_CARD]: ({ id, cardId, data }) =>
      requests.fetchWithAuthorisation(
        'patch',
        `/${CARDS_API}/${id}/source/${cardId}`,
        data,
      ),
    [DELETE_CUSTOMER_CARD]: ({ id, cardId }) =>
      requests.fetchWithAuthorisation(
        'delete',
        `/${CARDS_API}/${id}/source/${cardId}`,
      ),
  },
  processResult: {
    [CREATE_CUSTOMER_CARD]: CARD_NORMALISER.normaliseCreateCard,
    [DELETE_CUSTOMER_CARD]: CARD_NORMALISER.normaliseDeleteCard,
    [UPDATE_CUSTOMER_CARD]: CARD_NORMALISER.normaliseUpdateCard,
    [GET_CUSTOMER_CARDS]: CARD_NORMALISER.normaliseGetCard,
  },
  onSuccess: {},
  value: {},
  setValue: {},
};
