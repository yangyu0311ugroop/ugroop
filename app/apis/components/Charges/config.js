import { CHARGES_API, GET_CUSTOMER_CHARGES } from 'apis/constants';
import { CHARGES_NORMALISER } from 'apis/components/Charges/helpers';
import { requests } from 'utils/request';

export const CONFIG = {
  name: CHARGES_API,
  requests: {
    [GET_CUSTOMER_CHARGES]: ({ data }) =>
      requests.fetchWithAuthorisation(
        'get',
        `/${CHARGES_API}/?listOptions=${JSON.stringify(data)}`,
      ),
  },
  processResult: {
    [GET_CUSTOMER_CHARGES]: CHARGES_NORMALISER.normaliseListCharges,
  },
  onSuccess: {},
  value: {},
  setValue: {},
};
