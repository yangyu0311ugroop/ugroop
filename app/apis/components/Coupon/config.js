import { COUPON_API, GET_COUPON } from 'apis/constants';
import { requests } from 'utils/request';

export const CONFIG = {
  name: COUPON_API,
  requests: {
    [GET_COUPON]: id =>
      requests.fetchWithAuthorisation('get', `/${COUPON_API}/${id}`),
  },
};
