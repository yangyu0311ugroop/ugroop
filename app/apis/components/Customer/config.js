import { GET_CUSTOMER, UPDATE_CUSTOMER, CUSTOMER_API } from 'apis/constants';
import { requests } from 'utils/request';
import { CUSTOMER_NORMALISERS } from '../Subscription/normalisers';

export const CONFIG = {
  name: CUSTOMER_API,
  requests: {
    [GET_CUSTOMER]: ({ id }) =>
      requests.fetchWithAuthorisation('get', `/${CUSTOMER_API}/${id}`),
    [UPDATE_CUSTOMER]: ({ id, data }) =>
      requests.fetchWithAuthorisation('post', `/${CUSTOMER_API}/${id}`, data),
  },
  processResult: {
    [UPDATE_CUSTOMER]: CUSTOMER_NORMALISERS.updateCustomerSource,
  },
  onSuccess: {},
  value: {},
  setValue: {},
};
