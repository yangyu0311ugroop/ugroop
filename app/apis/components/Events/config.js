import { EVENTS_API, GET_EVENTS_BY_ID } from 'apis/constants';
import { EVENT_STORE } from 'appConstants';
import { EVENT_SCHEMA } from 'datastore/eventStore/schema';
import { requests } from 'utils/request';
import { normalize } from 'normalizr';

export const CONFIG = {
  name: EVENTS_API,

  requests: {
    [GET_EVENTS_BY_ID]: ({ ids }) =>
      requests.fetchWithAuthorisation('get', `/${EVENTS_API}?ids=[${ids}]`),
  },

  processResult: {
    [GET_EVENTS_BY_ID]: result => {
      const eventNormalize = normalize(result.eventList, EVENT_SCHEMA.events);
      const { entities } = eventNormalize;

      const flightNormalize = normalize(
        result.flightBookingList,
        EVENT_SCHEMA.flightBookings,
      );
      const { entities: flightEntities } = flightNormalize;

      const { events, attachments } = entities;
      const { flightBookings } = flightEntities;

      const flightBookingIds = flightNormalize.result;

      return {
        events,
        attachments,
        flightBookings,
        flightBookingIds,
      };
    },
  },

  setValue: {
    events: [EVENT_STORE, 'events'],
    attachments: [EVENT_STORE, 'attachments'],
    flightBookings: [EVENT_STORE, 'flightBookings'],
    flightBookingIds: [EVENT_STORE, 'flightBookingIds'],
  },

  processError: {},

  manuallySubscribe: true, // will not auto-subscribe to all values in `@@values`
};
