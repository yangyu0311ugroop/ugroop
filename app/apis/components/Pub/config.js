import NodeNormalisers from 'apis/components/Node/normalisers';
import { TEMPLATE_API_NORMALISERS } from 'apis/components/Template/normalisers';
import { TEMPLATE_TAB_UTILS } from 'apis/components/TemplateTab/utils';
import {
  BATCH_GET_PUB_TEMPLATE_TAB,
  GET_DETAILS,
  GET_PUB_TEMPLATE_EVENTS,
  GET_PUB_TEMPLATE_HEADER,
  GET_PUB_TEMPLATE_PEOPLE,
  GET_PUB_TEMPLATE_TAB,
  GET_PUB_TEMPLATE_TIMES,
  GET_PUB_TEMPLATE_TREE,
  PUB_API,
  PUB_CREATE_INTEREST,
} from 'apis/constants';

import {
  FILE_DATA_STORE,
  ORGANISATION_DATA_STORE,
  PHONE_DATA_STORE,
  TEMPLATE_MANAGEMENT_DATASTORE,
  USER_DATA_STORE,
  ATTACHMENT_DATASTORE,
} from 'appConstants';
import { EVENT_STORE_DATA_SELECTORS } from 'datastore/eventStore/selectors';
import { NODE_SCHEMA } from 'datastore/nodeStore/schema';
import { NODE_STORE_SELECTORS } from 'datastore/nodeStore/selectors';
import { ORGANISATION_SCHEMA } from 'datastore/orgStore/schema';
import { PERSON_STORE_SCHEMA } from 'datastore/personDataStore/schema';
import arrays from 'datastore/templateManagementStore/helpers/arrays';
import tabHelpers from 'datastore/templateManagementStore/helpers/tabs';
import { DATASTORE_UTILS } from 'datastore';
import { SET_VALUE } from 'ugcomponents/SnackBar/config';
import { compose } from 'redux';
import schema from 'datastore/templateManagementStore/schema';
import helpers from 'datastore/userStore/helpers';
import dotProp from 'dot-prop-immutable';
import _, { get } from 'lodash';
import { normalize } from 'normalizr';
import { requests } from 'utils/request';
import { NODE_API_UTILS } from '../Node/utils';
import { INVITATION_STORE_SELECTORS } from '../../../datastore/invitationStore/selectors';

export const convertChildren = () => children => arrays.convert(children);

export const CONFIG = {
  name: PUB_API,

  setValue: {
    user: [USER_DATA_STORE, 'users'],
    person: [USER_DATA_STORE, 'people'],
    photo: [FILE_DATA_STORE, 'files'],
    organisation: [ORGANISATION_DATA_STORE, 'organisations'],
    location: [ORGANISATION_DATA_STORE, 'locations'],
    files: [FILE_DATA_STORE, 'files'],
    nodes: NODE_STORE_SELECTORS.nodes,
    calculatedNodes: NODE_STORE_SELECTORS.calculatedNodes,
    events: EVENT_STORE_DATA_SELECTORS.events,
    flightBookings: EVENT_STORE_DATA_SELECTORS.flightBookings,
    flightBookingIds: EVENT_STORE_DATA_SELECTORS.flightBookingIds,
    id: [TEMPLATE_MANAGEMENT_DATASTORE, 'id'],
    shares: INVITATION_STORE_SELECTORS.shares,
    shareIds: INVITATION_STORE_SELECTORS.shareIds,
    userNodes: INVITATION_STORE_SELECTORS.userNodes,
    userNodeIds: INVITATION_STORE_SELECTORS.userNodeIds,
    personPhones: [PHONE_DATA_STORE, 'phones'],
    attachments: [ATTACHMENT_DATASTORE, 'attachments'],
    nodeSettings: NODE_STORE_SELECTORS.nodeSettings,
    ...SET_VALUE,
  },

  requests: {
    [GET_PUB_TEMPLATE_EVENTS]: ({ hashkey }) =>
      requests.fetchWithURL('get', `/${PUB_API}/template/${hashkey}/events`),
    [GET_PUB_TEMPLATE_PEOPLE]: ({ hashkey }) =>
      requests.fetchWithURL('get', `/${PUB_API}/template/${hashkey}/people`),
    [GET_PUB_TEMPLATE_TIMES]: ({ hashkey, ids }) => {
      if (Array.isArray(ids) && ids.length) {
        return requests.fetchWithURL(
          'get',
          `/${PUB_API}/template/${hashkey}/times?ids=${JSON.stringify(ids)}`,
        );
      }

      return undefined;
    },
    [GET_PUB_TEMPLATE_TREE]: ({ hashkey }) =>
      requests.fetchWithURL('get', `/${PUB_API}/template/${hashkey}/tree`),
    [GET_PUB_TEMPLATE_HEADER]: ({ hashkey }) =>
      requests.fetchWithURL('get', `/${PUB_API}/template/${hashkey}`),
    [GET_PUB_TEMPLATE_TAB]: ({ hashkey, id }) =>
      requests.fetchWithURL('get', `/${PUB_API}/template/${hashkey}/tab/${id}`),
    [BATCH_GET_PUB_TEMPLATE_TAB]: ({ hashkey, items }) =>
      Promise.all(
        items.map(async item =>
          requests.fetchWithURL(
            'get',
            `/${PUB_API}/template/${hashkey}/tab/${item.id}`,
          ),
        ),
      ),
    [PUB_CREATE_INTEREST]: ({ hashkey, data }) =>
      requests.fetchWithURL(
        'post',
        `/${PUB_API}/template/${hashkey}/interest`,
        data,
      ),
    [GET_DETAILS]: ({ userIds = [], organisationIds = [] }) => {
      if (!userIds.length && !organisationIds.length) {
        return {};
      }

      return requests.fetchWithURL(
        'get',
        `/${PUB_API}/details?userIds=${JSON.stringify(
          userIds,
        )}&organisationIds=${JSON.stringify(organisationIds)}`,
      );
    },
  },

  processResult: {
    [GET_PUB_TEMPLATE_TREE]: NodeNormalisers.getTree,
    [GET_PUB_TEMPLATE_EVENTS]: TEMPLATE_API_NORMALISERS.fetchEvents,
    [GET_PUB_TEMPLATE_PEOPLE]: helpers.normaliseGetPeople,
    [GET_PUB_TEMPLATE_HEADER]: (template, param) => {
      const origin = dotProp.set(template, 'children', convertChildren());

      const { entities, result } = normalize(origin, schema.templates);

      const { templates } = entities;
      const rest = templates;
      const { hashkeyDescription } = rest[origin.id];
      const getHashKeyDescription = {
        [origin.id]: {
          haskeyDescription: hashkeyDescription || null,
          hashkey: get(param, 'hashkey', ''),
        },
      };
      const { entities: nodeEntities } = normalize(
        origin,
        NODE_SCHEMA.singleNode,
      );

      const settings = NodeNormalisers.createTemplateSettings(
        template,
        nodeEntities,
      );

      const calculatedTabs = {
        [origin.id]: {
          calculated: {
            galleryId: NODE_API_UTILS.getGalleryId(origin),
            timelineId: NODE_API_UTILS.getTimelineId(origin),
            peopleTabId: NODE_API_UTILS.getTimelineId(origin),
          },
        },
      };
      const { photo, organisation, ...spread } = entities;

      return {
        ...spread,
        organisation: DATASTORE_UTILS.upsertObject(organisation),
        nodes: compose(
          DATASTORE_UTILS.upsertObject(nodeEntities.node),
          DATASTORE_UTILS.upsertObject(nodeEntities.eventNodes),
          DATASTORE_UTILS.upsertObject(calculatedTabs),
          DATASTORE_UTILS.upsertObject(getHashKeyDescription),
        ),
        files: compose(
          DATASTORE_UTILS.upsertObject(photo),
          DATASTORE_UTILS.upsertObject(nodeEntities.photo),
        ),
        id: result,
        nodeSettings: DATASTORE_UTILS.upsertObject(settings),
      };
    },
    [GET_PUB_TEMPLATE_TAB]: (result, payload) => {
      const { type, children, id, parentNodeId } = result;
      const tabChildrenData = tabHelpers.convertChildrenToArray(type, children);
      const normalised = normalize(tabChildrenData, NODE_SCHEMA.node);
      const entities = normalised.entities;
      const ids = normalised.result;
      const templateId = parentNodeId || payload.templateId;

      return {
        nodes: compose(
          DATASTORE_UTILS.upsertObject(entities.node),
          DATASTORE_UTILS.upsertObject(entities.eventNodes),
          DATASTORE_UTILS.upsertArray(`${id}.children`, ids),
        ),
        calculatedNodes: DATASTORE_UTILS.upsertArray(
          `${templateId}.calculated.events`,
          DATASTORE_UTILS.getObjectIds(entities.eventNodes),
        ),
        files: DATASTORE_UTILS.upsertObject(
          Object.assign(
            get(entities, 'photo', {}),
            get(entities, 'attachment', {}),
          ),
        ),
        attachments: DATASTORE_UTILS.upsertObject(
          Object.assign(get(entities, 'attachment', {})),
        ),
        ids,
      };
    },
    [BATCH_GET_PUB_TEMPLATE_TAB]: result =>
      TEMPLATE_TAB_UTILS.addMultipleNode(result),
    [PUB_CREATE_INTEREST]: (result = []) => {
      const id = get(result, '0.parentNodeId');
      const desc = get(result, '0.hashkeyDescription');
      const getHashKeyDescription = {
        [id]: {
          haskeyDescription: desc || null,
        },
      };
      return {
        nodes: DATASTORE_UTILS.upsertObject(getHashKeyDescription),
      };
    },
    [GET_DETAILS]: ({ persons = [], organisations = [] }) => {
      if (!persons.length && !organisations.length) return {};

      const { person, photo } = normalize(
        persons,
        PERSON_STORE_SCHEMA.peoples,
      ).entities;

      const { organisation, photo: orgPhoto, location } = normalize(
        organisations,
        ORGANISATION_SCHEMA.organisations,
      ).entities;

      return {
        person: DATASTORE_UTILS.upsertObject(person),
        organisation: DATASTORE_UTILS.upsertObject(organisation),
        location: DATASTORE_UTILS.upsertObject(location),
        photo: compose(
          DATASTORE_UTILS.upsertObject(photo),
          DATASTORE_UTILS.upsertObject(orgPhoto),
        ),
      };
    },
  },

  processError: error =>
    _.get(error, 'response.error.headers.error') || // 500 error
    _.get(error, 'response.error.message') || // 4xx errors
    error, // others

  manuallySubscribe: true, // will not auto-subscribe to all values in `@@values`
};
