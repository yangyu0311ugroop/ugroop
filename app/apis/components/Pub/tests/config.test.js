import {
  GET_PUB_TEMPLATE_EVENTS,
  GET_PUB_TEMPLATE_PEOPLE,
  GET_PUB_TEMPLATE_TIMES,
  GET_PUB_TEMPLATE_TREE,
  GET_PUB_TEMPLATE_TAB,
  GET_PUB_TEMPLATE_HEADER,
  BATCH_GET_PUB_TEMPLATE_TAB,
  PUB_CREATE_INTEREST,
  GET_DETAILS,
} from 'apis/constants';
import { requests } from 'utils/request';
import { CONFIG, convertChildren } from '../config';
import { NODE_API_UTILS } from '../../Node/utils';

jest.mock('dot-prop-immutable', () => ({
  set: () => ({ id: 1 }),
}));

describe('apis/Pub/config', () => {
  requests.fetchWithURL = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  describe('CONFIG', () => {
    it('exists', () => {
      expect(CONFIG).toBeDefined();
    });
  });

  describe('#setValue', () => {
    it('contains required properties', () => {
      expect(typeof CONFIG.setValue).toBe('object');
    });
  });

  describe('#requests', () => {
    it('still matches snapshot', () => {
      CONFIG.requests[GET_PUB_TEMPLATE_EVENTS]({ hashkey: 'abcd' });
      CONFIG.requests[GET_PUB_TEMPLATE_PEOPLE]({ hashkey: 'abcd' });
      CONFIG.requests[GET_PUB_TEMPLATE_TIMES]({ hashkey: 'abcd', ids: [1] });
      CONFIG.requests[GET_PUB_TEMPLATE_TIMES]({ hashkey: 'abcd', ids: [] });
      CONFIG.requests[GET_PUB_TEMPLATE_TREE]({ hashkey: 'abcd' });
      CONFIG.requests[GET_PUB_TEMPLATE_HEADER]({ hashkey: 'abcd' });
      CONFIG.requests[GET_PUB_TEMPLATE_TAB]({ hashkey: 'abcd' });
      CONFIG.requests[PUB_CREATE_INTEREST]({ hashkey: 'abcd', data: 'data' });
      CONFIG.requests[GET_DETAILS]({});
      CONFIG.requests[GET_DETAILS]({
        userIds: [10, 11],
        organisationIds: [22, 33],
      });
      expect(requests.fetchWithURL.mock.calls).toMatchSnapshot();
    });
  });

  describe('#processError', () => {
    it('processes correctly', () => {
      const errorMsg = 'Test error message';
      expect(
        CONFIG.processError({
          response: { error: { headers: { error: errorMsg } } },
        }),
      ).toEqual(errorMsg);
      expect(
        CONFIG.processError({ response: { error: { message: errorMsg } } }),
      ).toEqual(errorMsg);
      const otherError = { other: { format: errorMsg } };
      expect(CONFIG.processError(otherError)).toEqual(otherError);
    });
  });

  describe('processResult', () => {
    it('should process result of GET_PUB_TEMPLATE_HEADER', () => {
      NODE_API_UTILS.getGalleryId = jest.fn(() => 11);
      NODE_API_UTILS.getTimelineId = jest.fn(() => 12);
      const template = {
        content: 'content',
        id: 1,
        customData: {
          shortDescription: 'shortDescription',
        },
      };

      CONFIG.processResult[GET_PUB_TEMPLATE_HEADER](template);
    });
    it('should process result of GET_PUB_TEMPLATE_TAB', () => {
      const type = 'tabother';
      const id = 1;
      const children = {
        0: {
          content: 'content',
          type: 'activity',
          parentNodeId: 2,
        },
      };

      CONFIG.processResult[GET_PUB_TEMPLATE_TAB](
        { type, children, id },
        { templateId: 1 },
      );
    });
    it('should process the result of BATCH_GET_PUB_TEMPLATE_TAB', () => {
      const result = [
        { content: '1', id: 1 },
        { content: '2', id: 2 },
        { content: '3', id: 3 },
      ];
      expect(
        CONFIG.processResult[BATCH_GET_PUB_TEMPLATE_TAB](result, {}),
      ).toMatchSnapshot();
    });
    it('should process the result of PUB_CREATE_INTEREST', () => {
      const getHashKeyDescription = {
        1: {
          haskeyDescription: 'abcd' || null,
        },
      };
      CONFIG.processResult[PUB_CREATE_INTEREST](getHashKeyDescription);
      expect(typeof CONFIG.processResult[PUB_CREATE_INTEREST]()).toBe('object');
    });

    it('should process the result of GET_DETAILS', () => {
      const data = {
        persons: [
          { userId: 11, photo: [{ url: 'url' }], knownAs: 'Test 11' },
          { userId: 12, photo: [{ url: 'url2' }], knownAs: 'Test 12' },
        ],
        organisations: [
          {
            id: 22,
            photo: [{ url: 'url22' }],
            location: { location: 'location22' },
          },
          {
            id: 33,
            photo: [{ url: 'url33' }],
            location: { location: 'location33' },
          },
        ],
      };
      expect(CONFIG.processResult[GET_DETAILS](data)).toMatchSnapshot();
    });

    it('should process the result of GET_DETAILS', () => {
      expect(CONFIG.processResult[GET_DETAILS]({})).toMatchSnapshot();
    });
  });

  describe('convertChildren', () => {
    it('should convert children', () => {
      const children = { 1: '1', 2: '2', 3: '3' };
      convertChildren()(children);
    });
  });

  describe('BATCH_GET_PUB_TEMPLATE_TAB', () => {
    it('should call multiple fetchWithAuthorisation with particular parameter', () => {
      requests.fetchWithURL = jest.fn();
      const mockTabs = [
        {
          id: 1,
          type: 'tabtimeline',
        },
        {
          id: 2,
          type: 'tabtother',
        },
        {
          id: 3,
          type: 'tabother',
        },
      ];
      CONFIG.requests[BATCH_GET_PUB_TEMPLATE_TAB]({
        hashkey: 'hashkey',
        items: mockTabs,
      });
      expect(requests.fetchWithURL).toBeCalled();
    });
  });
});
