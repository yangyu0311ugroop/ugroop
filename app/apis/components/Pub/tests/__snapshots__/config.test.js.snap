// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`apis/Pub/config #requests still matches snapshot 1`] = `
Array [
  Array [
    "get",
    "/Pub/template/abcd/events",
  ],
  Array [
    "get",
    "/Pub/template/abcd/people",
  ],
  Array [
    "get",
    "/Pub/template/abcd/times?ids=[1]",
  ],
  Array [
    "get",
    "/Pub/template/abcd/tree",
  ],
  Array [
    "get",
    "/Pub/template/abcd",
  ],
  Array [
    "get",
    "/Pub/template/abcd/tab/undefined",
  ],
  Array [
    "post",
    "/Pub/template/abcd/interest",
    "data",
  ],
  Array [
    "get",
    "/Pub/details?userIds=[10,11]&organisationIds=[22,33]",
  ],
]
`;

exports[`apis/Pub/config processResult should process the result of BATCH_GET_PUB_TEMPLATE_TAB 1`] = `
Object {
  "files": [Function],
  "nodes": [Function],
  "result": Array [
    1,
    2,
    3,
  ],
}
`;

exports[`apis/Pub/config processResult should process the result of GET_DETAILS 1`] = `
Object {
  "location": [Function],
  "organisation": [Function],
  "person": [Function],
  "photo": [Function],
}
`;

exports[`apis/Pub/config processResult should process the result of GET_DETAILS 2`] = `Object {}`;
